src/Auth0.AuthenticationApi/Tokens/SignedDecoder.cs

```diff
@@ -0,0 +1,62 @@
+﻿using Microsoft.IdentityModel.Tokens;
+using System;
+using System.Collections.Generic;
+using System.IdentityModel.Tokens.Jwt;
+
+namespace Auth0.AuthenticationApi.Tokens
+{
+    internal abstract class SignedDecoder
+    {
+        readonly JwtSignatureAlgorithm signatureAlgorithm;
+        readonly JwtSecurityTokenHandler securityTokenHandler = new JwtSecurityTokenHandler();
+        readonly TokenValidationParameters validationParameters = new TokenValidationParameters
+        {
+            ValidateAudience = false,
+            ValidateIssuer = false,
+            ValidateActor = false,
```

Comment:
What's this?

Reply:
We're using Microsoft's library to perform the signature validation but there's no specific "ValidateSignature" function.  Instead we have to call the general Validation function and switch all the other types of validations off.